File: invision-backend/app/Enums/PaymentStatus.php
<?php

namespace App\Enums;

enum PaymentStatus: string
{
    case Pending = 'pending';
    case Paid = 'paid';
    case PartiallyPaid = 'partially_paid';
    case Overdue = 'overdue';
    case Refunded = 'refunded';

    public function label(): string
    {
        return match ($this) {
            self::Pending => 'Pending',
            self::Paid => 'Paid',
            self::PartiallyPaid => 'Partially Paid',
            self::Overdue => 'Overdue',
            self::Refunded => 'Refunded',
        };
    }

    public function color(): string
    {
        return match ($this) {
            self::Pending => 'yellow',
            self::Paid => 'green',
            self::PartiallyPaid => 'blue',
            self::Overdue => 'red',
            self::Refunded => 'gray',
        };
    }
}

File: invision-mobile/lib/src/core/enums/payment_status.dart
enum PaymentStatus {
  pending,
  paid,
  partiallyPaid,
  overdue,
  refunded;

  String get label => switch (this) {
    pending => 'Pending',
    paid => 'Paid',
    partiallyPaid => 'Partially Paid',
    overdue => 'Overdue',
    refunded => 'Refunded',
  };

  static PaymentStatus fromString(String value) {
    return switch (value) {
      'pending' => PaymentStatus.pending,
      'paid' => PaymentStatus.paid,
      'partially_paid' => PaymentStatus.partiallyPaid,
      'overdue' => PaymentStatus.overdue,
      'refunded' => PaymentStatus.refunded,
      _ => PaymentStatus.pending,
    };
  }
}
